api/utils.go

```diff
@@ -102,6 +102,28 @@ func txDataPresent(txHash types.Hash, cfg *config.Config, db *sql.DB, mempool bo
 	return present
 }
 
+func previousBlockPresent(input interface{}, cfg *config.Config, db *sql.DB, isTx bool) bool {
```

Comment:
I would also question the utility of having one function delineated by an isTx bool instead of two functions. There's only a couple of lines of code in common.